spring-integration-test/src/main/java/org/springframework/integration/test/context/MockIntegrationContext.java

```diff
@@ -112,15 +120,31 @@ public void instead(String pollingAdapterId, MessageSource<?> mockMessageSource,
 		instead(pollingAdapterId, mockMessageSource, SourcePollingChannelAdapter.class, "source", autoStartup);
 	}
 
-	private void instead(String endpointId, Object mock, Class<?> endpointClass, String property,
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler) {
+		instead(consumerEndpointId, mockMessageHandler, true);
+	}
+
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler, boolean autoStartup) {
+		instead(consumerEndpointId, mockMessageHandler, IntegrationConsumer.class, "handler", autoStartup);
+		if (mockMessageHandler instanceof MessageProducer) {
```

Comment:
Thank you, @Toparvion, for coming  back for review!

Let me try to answer to your concerns.

This PR is exactly about `MessageHandler` part which is possible in the middle/end of flow.
The Inbound Gateway is fully different story and can be covered via different approach.

So, here we deal with `MessageHandler` which is also a `MessageProducer`.
In Framework they are all of the `AbstractMessageProducingHandler` extensions.

So, this new `MockMessageHandler` is exactly for them, nothing more.

Since we **mock** here anyway that should be expectable that there is something mocking of our real components. There it should not be surprise when we tell `isntead("foo", mock)`, nothing is broken in the flow logic, but only we have a verification `MH` in the required place. So, I don't see reason (yet) to document the feature.